src/sas/sasgui/perspectives/fitting/gpu_options.py

```diff
@@ -0,0 +1,173 @@
+
+        help_id = wx.NewId()
+        help_btn = wx.Button(self, help_id, 'Help')
+        help_btn.SetToolTipString("Help on the GPU options")
+
+        reset_id = wx.NewId()
+        reset_btn = wx.Button(self, reset_id, 'Reset')
+        reset_btn.SetToolTipString("Restore initial settings")
+
+        self.Bind(wx.EVT_BUTTON, self.on_OK, accept_btn)
+        self.Bind(wx.EVT_BUTTON, self.on_reset, reset_btn)
+        self.Bind(wx.EVT_BUTTON, self.on_help, help_btn)
+
+        btn_sizer = wx.BoxSizer(wx.HORIZONTAL)
+        btn_sizer.Add((10, 20), 1) # stretchable whitespace
+        btn_sizer.Add(accept_btn, 0)
+        btn_sizer.Add((10, 20), 0) # non-stretchable whitespace
+        btn_sizer.Add(reset_btn, 0)
+        btn_sizer.Add((10, 20), 0) # non-stretchable whitespace
+        btn_sizer.Add(help_btn, 0)
+
+        # Add the button sizer to the main sizer.
+        self.vbox.Add(btn_sizer, 0, wx.EXPAND|wx.ALL, 10)
+
+        self.SetSizer(self.vbox)
+        self.vbox.Fit(self)
+        self.SetTitle("OpenCL options")
+        self.Centre()
+
+    def _get_clinfo(self):
+        """
+        Reading in information about available OpenCL infrastructure
+        :return:
+        """
+        clinfo = []
+        try:
+            import pyopencl as cl
+            platforms = cl.get_platforms()
+            p_index = 0
+            for platform in platforms:
+                d_index = 0
+                devices = platform.get_devices()
+                for device in devices:
+                    if len(devices) > 1 and len(platforms) > 1:
+                        combined_index = ":".join([str(p_index),str(d_index)])
+                    elif len(platforms) > 1:
+                        combined_index = str(p_index)
+                    else:
+                        combined_index = str(d_index)
+                    #combined_index = ":".join([str(p_index),str(d_index)]) \
+                    #    if len(platforms) > 1 else str(d_index)
+                    clinfo.append((combined_index, ":".join([platform.name,device.name])))
+                    d_index += 1
+                p_index += 1
+        except ImportError:
+            warnings.warn("pyopencl import failed. Using only CPU computations")
+
+        clinfo.append(("None","No OpenCL"))
+        return clinfo
+
+    def on_check(self, event):
+        """
+        Action triggered when button is selected
+        :param event:
+        :return:
+        """
+        selected_button = event.GetEventObject()
+        for btn in self.buttons:
+            if btn != selected_button:
+                btn.SetValue(0)
+        if selected_button.GetValue():
+            self.sas_opencl = self.option_button[selected_button.Name]
+        else:
+            self.sas_opencl = None
+
+    def on_OK(self, event):
+        """
+        Close window on accpetance
+        """
+        import sasmodels
+        #If statement added to handle Reset
+        if self.sas_opencl:
```

Comment:
Should save the selection somewhere in ~/.sasview/config, and reload it on sasview startup.  Otherwise you have to set it at the beginning of every session.